core/server/services/mega/mega.js

```diff
@@ -228,14 +243,24 @@ async function listener(emailModel, options) {
     }
 }
 
+const statusChangeListener = (emailModel, options) => {
+    const emailRetried = emailModel.wasChanged() && (emailModel.get('status') === 'pending') && (emailModel.previous('status') === 'failed');
+
+    if (emailRetried) {
+        listener(emailModel, options);
+    }
+};
+
 function listen() {
     common.events.on('email.added', listener);
+    common.events.on('email.edited', statusChangeListener);
```

Comment:
Don't think it's a good name - `statusChangeListener`. It is meant to indicate that the previously failed emails batch was marked as `pending` again (meaning we can resend it). Also the other method name `listener` is way to generic now. Probably will rename it to `pendingEmailHandler` :thinking: